Polish rpc docstring.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30069 Polish rpc docstring.**
* #30068 Add distributed optimizer section to distributed autograd design doc.

1) Fix rpc docstrings
2) Fix some links

Differential Revision: [D18588231](https://our.internmc.facebook.com/intern/diff/D18588231/)

Commit:
Polish rpc docstring.

1) Fix rpc docstrings
2) Fix some links

Differential Revision: [D18588231](https://our.internmc.facebook.com/intern/diff/D18588231/)

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -35,9 +35,15 @@ PyObject* rpc_init(PyObject* /* unused */) {
+          "WorkerInfo",
+          R"(
+  Encapsulates information of a worker in the system.
+)")
```
The indent of this right bracket is different from others

mrshenli reviewed (commented):
Could you please post a snapshot of the result?

Commit:
Update on "Polish rpc docstring."


1) Fix rpc docstrings
2) Fix some links

Differential Revision: [D18588231](https://our.internmc.facebook.com/intern/diff/D18588231/)

[ghstack-poisoned]

pritamdamania87 commented:
I've uploaded screenshots although it might be easier to patch the PR locally and view it in your browser.

<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 12 AM" src="https://user-images.githubusercontent.com/9958665/69173100-f7245a80-0ab3-11ea-8fe9-61336016a749.png">
<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 18 AM" src="https://user-images.githubusercontent.com/9958665/69173101-f7bcf100-0ab3-11ea-8918-380c89ab8f2e.png">
<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 24 AM" src="https://user-images.githubusercontent.com/9958665/69173106-f7bcf100-0ab3-11ea-9e49-6323ce1a7277.png">
<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 31 AM" src="https://user-images.githubusercontent.com/9958665/69173107-f8558780-0ab3-11ea-86ef-aa0e070ca9f5.png">
<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 36 AM" src="https://user-images.githubusercontent.com/9958665/69173109-f8558780-0ab3-11ea-9285-fa48360d3c53.png">
<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 42 AM" src="https://user-images.githubusercontent.com/9958665/69173112-f8558780-0ab3-11ea-94ef-924a2d73edff.png">
<img width="1680" alt="Screen Shot 2019-11-19 at 10 01 48 AM" src="https://user-images.githubusercontent.com/9958665/69173113-f8ee1e00-0ab3-11ea-8d63-54eb60f7ac58.png">

mrshenli commented on docs/source/notes/distributed_autograd.rst:
```diff
@@ -3,9 +3,6 @@
 
 .. _distributed-autograd-design:
 
-.. warning::
```
Removing this because it is already mentioned in the API doc?

mrshenli commented on torch/distributed/rpc/api.py:
```diff
@@ -235,12 +233,12 @@ def rpc_async(to, func, args=None, kwargs=None):
 
     Arguments:
-        to (int or str): id or name of the destination worker.
+        to (int or str or WorkerInfo): id or name of the destination worker.
```
It's prior to this PR, but it seems this cannot be int, as the API is not exposed in `RpcAgent`.

mrshenli commented on torch/distributed/rpc/api.py:
```diff
@@ -253,21 +251,17 @@ def rpc_async(to, func, args=None, kwargs=None):
         >>> import torch.distributed.rpc as rpc
-        >>> dist.init_process_group(backend='gloo', rank=0, world_size=2)
-        >>> rpc.init_rpc("worker0")
+        >>> rpc.init_rpc("worker0", self_rank=0, world_size=2)
```
`self_rank` has been renamed to `rank` in #29747

mrshenli commented on torch/distributed/rpc/api.py:
```diff
@@ -253,21 +251,17 @@ def rpc_async(to, func, args=None, kwargs=None):
         >>> import torch.distributed.rpc as rpc
-        >>> dist.init_process_group(backend='gloo', rank=1, world_size=2)
-        >>> rpc.init_rpc("worker1")
+        >>> rpc.init_rpc("worker1", self_rank=0, world_size=2)
```
same here

mrshenli commented on torch/distributed/rpc/api.py:
```diff
@@ -211,18 +213,14 @@ def rpc_sync(to, func, args=None, kwargs=None):
         >>> import torch.distributed.rpc as rpc
-        >>> dist.init_process_group(backend='gloo', rank=1, world_size=2)
-        >>> rpc.init_rpc("worker1")
+        >>> rpc.init_rpc("worker1", self_rank=0, world_size=2)
```
self_rank -> rank. There are multiple places above.

mrshenli reviewed (approved):
API change looks good to me. Thanks!

One test failure is irrelevant:

```
Nov 19 19:31:52 ======================================================================
Nov 19 19:31:52 FAIL: test_mnist_cuda (jit.test_models.TestModels)
Nov 19 19:31:52 ----------------------------------------------------------------------
Nov 19 19:31:52 Traceback (most recent call last):
Nov 19 19:31:52   File "/var/lib/jenkins/workspace/test/common_utils.py", line 610, in wrapper
Nov 19 19:31:52     method(*args, **kwargs)
Nov 19 19:31:52   File "/var/lib/jenkins/workspace/test/common_utils.py", line 494, in __exit__
Nov 19 19:31:52     self.name, after - before, i))
Nov 19 19:31:52   File "/var/lib/jenkins/workspace/test/common_utils.py", line 815, in assertEqual
Nov 19 19:31:52     super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
Nov 19 19:31:52 AssertionError: 63488 not less than or equal to 1e-05 : jit.test_models.TestModels.test_mnist_cuda leaked 63488 bytes CUDA memory on device 0
```

Another test failure will disappear after a rebase:

```
19:26:47 ======================================================================
19:26:47 ERROR: test_rpc_return_rref (__main__.RpcTestWithFork)
19:26:47 ----------------------------------------------------------------------
19:26:47 Traceback (most recent call last):
19:26:47   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 130, in wrapper
19:26:47     self._join_processes(fn)
19:26:47   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 211, in _join_processes
19:26:47     self._check_return_codes(elapsed_time)
19:26:47   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 230, in _check_return_codes
19:26:47     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
19:26:47 RuntimeError: Process 0 terminated or timed out after 300.2686879634857 seconds
```

BTW, I use chrome plugin to take snapshot of the entire webpage ([e.g](https://chrome.google.com/webstore/detail/take-webpage-screenshots/mcbpblocgmgfnpjjppndjkmgjaogfceg?hl=en))

Commit:
Update on "Polish rpc docstring."


1) Fix rpc docstrings
2) Fix some links

Differential Revision: [D18588231](https://our.internmc.facebook.com/intern/diff/D18588231/)

[ghstack-poisoned]

Commit:
Update on "Polish rpc docstring."


1) Fix rpc docstrings
2) Fix some links

Differential Revision: [D18588231](https://our.internmc.facebook.com/intern/diff/D18588231/)

[ghstack-poisoned]
